ui/StatusQ/tests/TestCore/TestUtils/tst_SubscriptionBroker.qml

```diff
@@ -0,0 +1,293 @@
+        verify(subscription.subscriptionId != "", "subscription should have an id")
+
+        subscriptionBroker.subscribe(subscription)
+        compare(qtOuput.qtOuput().length, 0, `No output expected. Found:\n"${qtOuput.qtOuput()}"\n`)
+        compare(subscriptionBroker.subscribedSignalSpy.count, 1, "subscribed signal should be emitted")
+        compare(subscriptionBroker.subscribedSignalSpy.signalArguments[0][0], subscription.subscriptionId, "subscribed signal should be emitted with the subscription id")
+        compare(subscriptionBroker.unsubscribedSignalSpy.count, 0, "unsubscribed signal should not be emitted")
+        compare(subscriptionBroker.requestSignalSpy.count, 0, "request signal should not be emitted. Subscription is inactive by default. Broker is inactive by default.")
+        
+        subscriptionBroker.unsubscribe(subscription.subscriptionId)
+        compare(subscriptionBroker.subscribedSignalSpy.count, 1, "subscribed signal should not be emitted for unsunbscribe")
+        compare(subscriptionBroker.unsubscribedSignalSpy.count, 1, "unsubscribed signal should be emitted")
+        compare(subscriptionBroker.unsubscribedSignalSpy.signalArguments[0][0], subscription.subscriptionId, "unsubscribed signal should be emitted with the subscription id")
+        compare(subscriptionBroker.requestSignalSpy.count, 0, "request signal should not be emitted")
+    }
+
+    function test_periodic_request_one_subscription() {
+        const subscriptionBroker = createTemporaryObject(subscriptionBrokerComponent, testCase)
+        const subscription = createTemporaryObject(subscriptionComponent, testCase, {topic: "topic1", isReady: true, notificationInterval: 50})
+        
+        //Enable broker and subscription
+        //Verify that request signal is emitted after subscription
+        subscriptionBroker.active = true
+        subscriptionBroker.subscribe(subscription)
+        compare(subscriptionBroker.subscribedSignalSpy.count, 1, "subscribed signal should be emitted")
+        compare(subscriptionBroker.requestSignalSpy.count, 1, "request signal should be emitted")
+        compare(subscriptionBroker.requestSignalSpy.signalArguments[0][0], subscription.topic, "request signal should be emitted with the subscription topic")
+
+        //Verify that request signal is emitted after notificationInterval
+        //The broker expects a response before sending another request
+        subscriptionBroker.response(subscription.topic, "responseAsString")
+        compare(subscription.response, "responseAsString", "subscription response should be updated")
+
+        //first interval - coeck for one request every 50ms:
+        helper.resetSpies(subscriptionBroker)
+        compare(subscriptionBroker.requestSignalSpy.count, 0, "request signal should not be emitted")
+        wait(100)/*50ms error margin*/
+        compare(subscriptionBroker.requestSignalSpy.count, 1, "request signal should be emitted after 50ms")
```

Comment:
There are two things that imo can be improved here:
- I would resign from calling `resetSpies()` bc at least in theory it may be called already after the `request` signal emission. Comparing to just `2` is better option here.
- instead of calling `wait` and then `compare`, its more reliable (and gives faster tests execution) to use `tryCompare`. When we are not blocking for ~100ms but only ~50ms bc `tryCompare` will return immediately after property change.

Reply:
> I would resign from calling resetSpies() bc at least in theory it may be called already after the request signal emission. Comparing to just 2 is better option here.

The initial implementation was without resetSpies() and I found it hard to read the code. It's difficult to understand why a signalCount should be `3` for example. You need to keep in mind the previous value and then see if he signal count should or should not be changed.
As opposed to always read the test steps with 0 signals in mind.

> instead of calling wait and then compare, its more reliable (and gives faster tests execution) to use tryCompare. When we are not blocking for ~100ms but only ~50ms bc tryCompare will return immediately after property change.

Same here. I've been using it initially until I found it won't work properly for some comparison types.
Eg: Can't validate cases where we expect no change for the value under test (unchanged signal count after x ms); Can't detect defects where there are more requests than needed - in this case the tryCompare will pass when the expected value is hit. Going over this value is just ignored.

Reply:
Yeah, your points are definitely right. But on the other hand now you need to manage timeouts manually. There is comment explaining why 100, we could easily add comment explaining why 3, instead. Regarding observing no change, right, there is no way to use `wait`. In this case it's justified. Many changes - wait will hide that problem as well, right?

Anyway, the only real problem is delaying tests unnecessarily and risking race conditions. But it should work in general. So up to you which option to chose :)

Reply:
Yeah, we could easily (under high CPU server load) run into situations when even 500ms wouldn't be enough

Reply:
> Yeah, we could easily (under high CPU server load) run into situations when even 500ms wouldn't be enough

Not sure if it's that easy to run into these situations, but definitely possible. I mean we only need one loop every 50 ms and the loop is quite light. Let's see how it goes. If it's going to cause troubles we'll learn something about qml Timers and it's easy to make it optional.

> Yeah, your points are definitely right. But on the other hand now you need to manage timeouts manually. There is comment explaining why 100, we could easily add comment explaining why 3, instead. Regarding observing no change, right, there is no way to not use `wait`. In this case it's justified. Many changes - wait will hide that problem as well, right?
> 
> Anyway, the only real problem is delaying tests unnecessarily and risking race conditions. But it should work in general. So up to you which option to chose :)

I think you're right after all. I've used `wait` only where we're observing no change and tryVerify where we're expecting a change. The execution time hasn't changed, but it's more natural.
`resetSpies` is also removed 👍